addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -181,6 +182,18 @@ def _compute_valid(self):
             else:
                 holiday_type.has_valid_allocation = True
 
+    @api.constrains('requires_allocation')
+    def check_allocation_requirement_edit_validity(self):
+        """ Changing the allocation requirement on a leave type which is already in use is forbidden because
+        Odoo tracks every leave of that type, even those taken prior to switching to 'Allocation Required', as
+        needing allocation which messes with the remaining days to allocate.
+
+        Raises:
+            UserError: If leaves of the given type are found
+        """
```

Comment:
The message raised is sufficiently explicit, no need for that comment

Reply:
:+1: